vizro-core/src/vizro/models/_navigation/nav_item.py

```diff
@@ -0,0 +1,83 @@
+from __future__ import annotations
+
+import itertools
+from typing import TYPE_CHECKING, Optional
+
+import dash
+import dash_bootstrap_components as dbc
+from dash import html
+from pydantic import PrivateAttr, validator
+
+from vizro._constants import STATIC_URL_PREFIX
+from vizro.models import VizroBaseModel
+from vizro.models._models_utils import _log_call
+from vizro.models._navigation._navigation_utils import _validate_pages
+from vizro.models.types import NavigationPagesType
+
+if TYPE_CHECKING:
+    from vizro.models._navigation.accordion import Accordion
+
+
+class NavItem(VizroBaseModel):
+    """Icon to be used in Navigation Panel of Dashboard.
+
+    Args:
+        tooltip (Optional[str]): Text to be displayed in the tooltip on icon hover.
+        image (Optional[str]): URI (relative or absolute) of the embeddable content.
+        pages (Optional[NavigationPagesType]): See [NavigationPagesType][vizro.models.types.NavigationPagesType].
+                Defaults to `None`.
+    """
+
+    tooltip: Optional[str]
+    image: Optional[str]
+    pages: Optional[NavigationPagesType] = None
+    _selector: Accordion = PrivateAttr()
```

Comment:
Just to summarise what we've discussed w/ J:

We will have:

```
icon: Optional[str]
text: Optional[str]
tooltip: Optional[str]
max_text_length: int = 12 (let's see what looks good)
pages: NavigationPagesType
selector: Accordion
```

Long-term behaviour should be:
- **icon:** If None provide default image (but only on vertical), if a user sets icon = "False" or some other other falsy option, they can turn off our defaults
- **text:** If text goes above our max_text_length, it will be truncated to XXXX... and we should automatically populate the tooltip with the full text length
- **tooltip:** No special logic except for automatic population based on text length
- **max_text_length:** If a user wants to exceed the max text length, they can set it here in which case the text should wrap automatically (vertical alignment). For the horizontal alignment, we might just set a different max_text_length as default or None?

Reply:
Unresolving this for visibility for the other reviewer

Reply:
Behaviour and argument explanation added to the pr description.